src/schema.rs

```diff
@@ -212,4 +212,10 @@ pub enum StringFormat {
     Password,
     Byte,
     Binary,
+    Uuid,
+    Email,
+    Uri,
+    Hostname,
+    Ipv4,
+    Ipv6,
```

Comment:
I see this in the OpenAPI 3.0.3 spec:

> Formats such as "email", "uuid", and so on, MAY be used even though undefined by this specification.

I see "email" and "uuid" referenced (as well as byte, binary, date, and date-time), but I don't see uri, hostname, ipv4, ipv6. Why do you think we should add these? It seems like we already implicitly support these values via `VariantOrUnknownOrEmpty`

My instinct is to omit anything that's not explicitly defined by the spec.

Reply:
Well as they say "and so on". You can see them here: https://swagger.io/docs/specification/data-models/data-types/#string
The issue with other is that is does not serialize properly, because it will serialize to "other" right? What about other becoming other(String). And then we can serialize the String within so you get back exactly what you had?

Reply:
I may not be understanding what you're saying because it seems to serialize as the string you provider:

```rust
use openapiv3::*;

fn main() {
    let spec = Schema {
        schema_data: SchemaData::default(),
        schema_kind: SchemaKind::Type(Type::String(StringType {
            format: VariantOrUnknownOrEmpty::Unknown("ipv6".to_string()),
            pattern: None,
            enumeration: Vec::new(),
            min_length: None,
            max_length: None,
        })),
    };

    println!("{}", serde_json::to_string_pretty(&spec).unwrap());
}
```

```
$ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/openapiv3-test`
{
  "type": "string",
  "format": "ipv6"
}
```

Reply:
doh... I was so focused on the "::Other" I missed the fact that that the variant was another enum..... sorry for wasting your time -.-

Reply:
I've updated the commit with the skip stuff